src/custom/hooks/web3.ts

```diff
@@ -85,9 +85,18 @@ export function useEagerConnect() {
 
   // if the connection worked, wait until we get confirmation of that to flip the flag
   useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined
+
     if (active) {
       setTried(true)
+    } else {
+      timeout = setTimeout(() => {
+        localStorage.removeItem(STORAGE_KEY_LAST_PROVIDER)
```

Comment:
I don't get it. Why do we need to add this? maybe a comment would help to understand.

Reply:
The last provider is [stored before](https://github.com/gnosis/cowswap/blob/develop/src/custom/hooks/web3.ts#L27) the unload component.


After a page reload when trying to reconnect a wallet from an external provider such as **Fortmatic** or **CoinbaseWallet** we have no way of knowing if the session was cut off. 

The timeout would be the time we wait while the external provider reconnects the session, otherwise we assume that it has been tried, there is no `lastProvider` and the user can open `walletModals`.

Reply:
Thinking about it, maybe it makes more sense that this timeout logic is inside the [reconnectProvider](https://github.com/gnosis/cowswap/blob/develop/src/custom/hooks/web3.ts#L55) and this way it is only executed when there is a **lastProvider**.

I can do this together with #2515 but let me know if you have any suggestion on this.